pkg/kv/txn_interceptor_span_refresher.go

```diff
@@ -137,15 +135,6 @@ func (sr *txnSpanRefresher) SendLocked(
 			sr.refreshSpansBytes = 0
 		}
 	}
-	// If the transaction will retry and the refresh spans are
-	// exhausted, return a non-retryable error indicating that the
-	// transaction is too large and should potentially be split.
-	// We do this to avoid endlessly retrying a txn likely refail.
-	if sr.refreshInvalid && (br.Txn.OrigTimestamp != br.Txn.Timestamp) {
-		return nil, roachpb.NewErrorWithTxn(
-			errors.New("transaction is too large to complete; try splitting into pieces"), br.Txn,
```

Comment:
Note that this was added when we had workloads which were creating massive transactions as a result of executing statements like `DELETE FROM d.t WHERE timestamp > x`. These would retry endlessly because they could never finish so long as the timestamp was moved forward sufficiently. I would not be in favor of removing this protection. Perhaps you could enhance it instead.